### Importing Needed Module

In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split

### Reading Data

In [2]:
#Loading <data andrenaming columns
mushroom = pd.read_csv("agaricus-lepiota.data")
                      
mushroom.head()

p  x  s  n  t p.1  f  c n.1  k  ... s.2  w w.1 p.2 w.2  o p.3 k.1 s.3  u
0  e  x  s  y  t   a  f  c   b  k  ...   s  w   w   p   w  o   p   n   n  g
1  e  b  s  w  t   l  f  c   b  n  ...   s  w   w   p   w  o   p   n   n  m
2  p  x  y  w  t   p  f  c   n  n  ...   s  w   w   p   w  o   p   k   s  u
3  e  x  s  g  f   n  f  w   b  k  ...   s  w   w   p   w  o   e   n   a  g
4  e  x  y  y  t   a  f  c   b  n  ...   s  w   w   p   w  o   p   k   n  g

[5 rows x 23 columns]

### Cleaning Data

In [3]:
 #Renaming columns
names = ["classes", "cap-shape", "cap-surface",
         "cap-color", "bruises", "odor",
         "gill-attachment", "gill-spacing", "gill-size",
         "gill-color", "stalk-shape", "stalk-root",
         "stalk-surface-above-ring", "stalk-surface-below-ring", "stalk-color-above-ring",
         "stalk-color-below-ring", "veil-type", "veil-color",
         "ring-number", "ring-type", "spore-print-color",
         "population", "habitat"]
mushroom.columns = names
mushroom.head(1)

classes cap-shape cap-surface cap-color bruises odor gill-attachment  \
0       e         x           s         y       t    a               f   

  gill-spacing gill-size gill-color  ... stalk-surface-below-ring  \
0            c         b          k  ...                        s   

  stalk-color-above-ring stalk-color-below-ring veil-type veil-color  \
0                      w                      w         p          w   

  ring-number ring-type spore-print-color population habitat  
0           o         p                 n          n       g  

[1 rows x 23 columns]

In [4]:
#Checking <data information
mushroom.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8123 entries, 0 to 8122
Data columns (total 23 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   classes                   8123 non-null   object
 1   cap-shape                 8123 non-null   object
 2   cap-surface               8123 non-null   object
 3   cap-color                 8123 non-null   object
 4   bruises                   8123 non-null   object
 5   odor                      8123 non-null   object
 6   gill-attachment           8123 non-null   object
 7   gill-spacing              8123 non-null   object
 8   gill-size                 8123 non-null   object
 9   gill-color                8123 non-null   object
 10  stalk-shape               8123 non-null   object
 11  stalk-root                8123 non-null   object
 12  stalk-surface-above-ring  8123 non-null   object
 13  stalk-surface-below-ring  8123 non-null   object
 14  stalk-color-above-ring  

* There 8123 entries.
* And there are no missing values.
* Our data is in type string(Object),we may have to convert it to integer or float type.  
* We have 23 columns altogether.

In [5]:
#Checking the value counts of both edible and poisonous mushrooms in classes
edible = mushroom[mushroom.classes == "e"]
poisonous = mushroom[mushroom.classes == "p"]
print("Number of edible mushrooms =",len(edible))
print("Number of poisonous mushrooms =",len(poisonous))


Number of edible mushrooms = 4208
Number of poisonous mushrooms = 3915
